### PR TITLE
20240921 ssh keys

### PR DESCRIPTION
**Pull request by Firstyear**
# Change summary

- Adds support for unix pw and ssh updates in the credential update session
- In future we will remove the alternate methods to set this so we can unifi account policy here.

Fixes #

Checklist

- [ x ] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
I probably need to add book command updates with this ....

File: tools/cli/src/cli/person.rs [hunk 1]
```diff
@@ -1267,6 +1383,58 @@ fn display_status(status: CUStatus) {
...
+                print!("{}", cred_detail);
+            } else {
+                println!("  unable to modify - access denied");
+            }
+        }
+        CUCredState::AccessDeny => {
+            println!("  unable to modify - access denied");
+        }
+        CUCredState::PolicyDeny => {
+            println!("  unable to modify - account does not have posix attributes");
```
Comment (yaleman) [inline tools/cli/src/cli/person.rs:1406]:
```suggestion
            println!("  unable to modify - account does not have POSIX attributes");
```

File: tools/cli/src/cli/person.rs [hunk 2]
```diff
@@ -1458,6 +1626,57 @@ async fn credential_update_exec(
...
+                        ClientErrorHttp(_, Some(PasswordQuality(feedback)), _) => {
+                            eprintln!("Password was not secure enough, please consider the following suggestions:");
+                            for fb_item in feedback.iter() {
+                                eprintln!(" - {}", fb_item)
+                            }
+                        }
+                        _ => eprintln!("An error occurred -> {:?}", e),
+                    }
+                } else {
+                    println!("Successfully reset unix password.");
```
Comment (yaleman) [inline tools/cli/src/cli/person.rs:1656]:
```suggestion
                    println!("Successfully reset UNIX password.");
```

File: tools/cli/src/cli/person.rs [hunk 3]
```diff
@@ -1458,6 +1626,57 @@ async fn credential_update_exec(
...
+                }
+            }
+
+            CUAction::UnixPasswordRemove => {
+                if Confirm::new()
+                    .with_prompt("Do you want to remove your unix password?")
```
Comment (yaleman) [inline tools/cli/src/cli/person.rs:1662]:
```suggestion
                    .with_prompt("Do you want to remove your UNIX password?")
```

File: tools/cli/src/cli/person.rs [hunk 4]
```diff
@@ -1458,6 +1626,57 @@ async fn credential_update_exec(
...
+                    if let Err(e) = client
+                        .idm_account_credential_update_unix_remove(&session_token)
+                        .await
+                    {
+                        eprintln!("An error occurred -> {:?}", e);
+                    } else {
+                        println!("success");
+                    }
+                } else {
+                    println!("unix password was NOT removed");
```
Comment (yaleman) [inline tools/cli/src/cli/person.rs:1675]:
```suggestion
                    println!("UNIX password was NOT removed");
```

**Review by yaleman — CHANGES_REQUESTED** [review]

**Review by yaleman — APPROVED** [review]

**Review by yaleman — APPROVED** [review]

**Pull request merged**